src/functions.php

```diff
@@ -309,18 +333,8 @@ function filterMatches($attribute, $filterValues): bool
     }
 
     foreach ($filterValues as $filter) {
-        if (is_array($attribute)) {
-            // check if any attribute matches
-            foreach ($attribute as $childAttribute) {
-                if ($childAttribute === $filter) {
-                    return true;
-                }
-            }
-        } else {
-            // check if simple attribute matches
-            if ($attribute === $filter) {
-                return true;
-            }
+        if (strpos($attribute, $filter) !== false) {
```

Comment:
Is there a test for this change?

Reply:
Only paractic tests with real data back then. To be honest, this change is so long ago that I can not remember what made me shorten it. It took quite a long time for me to understand your coding. If I look at it now, I'm back in the beginning

Reply:
The original change allowed both arrays and strings as attributes. If there is no reason to not do or need this and no covering test case I'd propose to not change it.

Reply:
I walked again through the whole filter functions again.
I found a few bugs (in dissolveGroups() and config.example.php-> filters), deviations (in filter ()) and optimizations (in filtersMatch()). The latter makes filterMatches() superfluous.
Thus the resulting commit contains a few more changes!

Since we do not yet have a testcase for the filter function, it will take some time until I get it correctly defined on top :(